facturacion_mexico/facturacion_fiscal/utils.py

```
@@ -34,6 +34,56 @@ def get_invoice_uuid(sales_invoice_name):
+			row = frappe.db.get_value("Factura Fiscal Mexico", ffm_name, ["status", "fm_uuid"], as_dict=True)
+			if row and (row.fm_uuid or "").strip() and row.status in _FOLIO_VIGENTE_STATES:
+				uuid = row.fm_uuid.strip()
+
+		# Escribir solo si cambia (idempotencia, evita writes innecesarios)
+		current = frappe.db.get_value("Sales Invoice", sales_invoice_name, "fm_folio_fiscal") or ""
+		if current != uuid:
+			frappe.db.set_value(
+				"Sales Invoice", sales_invoice_name, "fm_folio_fiscal", uuid, update_modified=False
+			)
+
+		return uuid
```

Ruff (0.15.20)

[warning] Consider moving this statement to an `else` block

(TRY300)